src/validations/R/clade_prop_sum_one.R

```diff
@@ -0,0 +1,158 @@
+clade_prop_sum_one <- function(tbl, file_path) {
+  tbl <- tbl[tbl[["output_type"]] == "sample", ]
+  error_object <- check_sum_one(tbl)
+  check <- is.null(error_object)
+
+  if (check) {
+    details <- NULL
+    error_object <- NULL
+  } else {
+    n <- nrow(error_object)
+    bad_msg <- c("\fThere {?was/were} {.strong {.val {n}} task{?s} with values that did not sum to one}:",
+      cheap_kable(error_object)
+    )
+    # Wrapping 
+    # https://cli.r-lib.org/reference/inline-markup.html#wrapping
+    details <- cli::format_inline(paste(bad_msg, collapse = "\f"))
+  }
+
+  hubValidations::capture_check_cnd(
+    check = check,
+    error = FALSE,
+    file_path = file_path,
+    msg_subject = "All clade proportions in {.var value} columns",
+    msg_verbs = c("", "do not"),
+    msg_attribute = "{.emph sum to one} for each unique modeling task.",
+    error_object = error_object,
+    details = details
+  )
+}
+
+
+ALL_ONE <- function(dat) isTRUE(all.equal(sum(dat), 1))
```

Comment:
We want to add in a numerical tolerance here for being within +/- 0.001 of 1 (this specific tolerance was chosen more or less arbitrarily but it is ["codified" as of now](https://github.com/reichlab/variant-nowcast-hub/blob/bdfac214eb78219fc7c5cc4a32f4594540e78791/model-output/README.md?plain=1#L31)). We could change the tolerance if needed, although I think it's a reasonable balance between pragmatism and precision here.

It is likely that for some specific prediction task calculations the sums would not exactly equal 1 due to machine precision. We don't want it to throw an error in this case. I think the above code does not protect against that necessarily?

In other cases, a model may just not be correctly computing sums, in which case the errors are likely to be larger than 0.001. In these cases we definitely want errors

Reply:
the [`all.equal()` method for numeric](https://rdrr.io/r/base/all.equal.html) does protect against numerical precision errors (with a tolerance set to roughly 1.49e-8), but if the tolerance is expected to be 1e-3, then I can certainly set that.

Reply:
the deed is done.